public/app/plugins/panel/trend/TrendPanel.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useMemo } from 'react';
+
+import { FieldType, PanelProps } from '@grafana/data';
+import { config, PanelDataErrorView } from '@grafana/runtime';
+import { KeyboardPlugin, TimeSeries, TooltipDisplayMode, TooltipPlugin, usePanelContext } from '@grafana/ui';
+import { findFieldIndex } from 'app/features/dimensions';
+
+import { ContextMenuPlugin } from '../timeseries/plugins/ContextMenuPlugin';
+import { prepareGraphableFields, regenerateLinksSupplier } from '../timeseries/utils';
+
+import { PanelOptions } from './panelcfg.gen';
+
+export const TrendPanel = ({
+  data,
+  timeRange,
+  timeZone,
+  width,
+  height,
+  options,
+  fieldConfig,
+  replaceVariables,
+  id,
+}: PanelProps<PanelOptions>) => {
+  const { sync } = usePanelContext();
+
+  const info = useMemo(() => {
+    if (data.series.length > 1) {
+      return {
+        warning: 'Only one frame is supported, consider adding a join transformation',
+        frames: data.series,
+      };
+    }
+
+    let frames = data.series;
+    let xFieldIdx: number | undefined;
+    if (options.xField) {
+      xFieldIdx = findFieldIndex(frames[0], options.xField);
+      if (xFieldIdx == null) {
+        return {
+          warning: 'Unable to find field: ' + options.xField,
+          frames: data.series,
+        };
+      }
+    } else {
+      // first number field
+      // Perhaps we can/should support any ordinal rather than an error here
+      xFieldIdx = frames[0].fields.findIndex((f) => f.type === FieldType.number);
+      if (xFieldIdx === -1) {
+        return {
+          warning: 'No numeric fields found for X axis',
+          frames,
+        };
+      }
+    }
+
+    // Make sure values are ascending
+    if (xFieldIdx != null) {
+      const field = frames[0].fields[xFieldIdx];
+      if (field.type === FieldType.number) {
+        // we may support ordinal soon
+        let last = Number.NEGATIVE_INFINITY;
+        const values = field.values.toArray();
+        for (let i = 0; i < values.length; i++) {
+          const v = values[i];
+          if (last > v) {
+            return {
+              warning: `Values must be in ascending order (index: ${i}, ${last} > ${v})`,
```

Comment:
@leeoniya -- I'm adding this explicit check here... seems better that we block the weird behaviors that almost work from the start.  Then we can avoid the issue with timeseries, where we have to sort implicitly 😬

Reply:
buuuhhhhttt...

https://github.com/grafana/grafana/blob/afc9925dbfa55886cc85a9a2106ed9892d7df239/packages/grafana-data/src/transformations/transformers/joinDataFrames.ts#L355